### DISCUSSION TITLE
Function in PSADT

### DISCUSSION THREAD

**Discussion by FRacine** [PSAppDeployToolkit/PSAppDeployToolkit#1133] [Q&A] [answered]
I am looking for the functions and don't see them.
Where is the write-log function?

**Comment by sintaxasn** [comment]
If you intend to use v3 compatibility mode then Write-Log will be found in AppDeployToolkitMain.ps1 in the v3 template. If you intend to use native v4, it's Write-ADTLogEntry in the PSAppDeployToolkit module. 

We'll be pushing through extensive updates to the documentation over the course of the next few days. Please bear with us.

**Reply by FRacine** [reply]
> If you intend to use v3 compatibility mode then Write-Log will be found in AppDeployToolkitMain.ps1 in the v3 template. If you intend to use native v4, it's Write-ADTLog in the main module.

You mean Write-ADTLogEntry not Write-ADTlog?!
In V4 what is the main module?

**Reply by sintaxasn** [reply, marked as answer]
Apologies. I'm literally trying to reply before a flight takes off so right now and I missed autocorrect. Yes, Write-ADTLogEntry.

There is only one module. Poor choice of words on my behalf.
